policy/lib/rule_data.rego

```diff
@@ -93,18 +93,33 @@ rule_data_defaults := {
 rule_data(key_name) := value if {
 	# Expected to be defined under `configuration.rule_data` in the
 	# ECP configuration data being used when EC is run.
+	time.parse_ns("2024-01-01T00:00:00Z", rule_data_effective_on(data.rule_data__configuration__)) < time.now_ns()
```

Comment:
The format follows golang format string, this is what we want to use to parse [RFC3339](https://pkg.go.dev/time#pkg-constants) formatted strings

```suggestion
	time.parse_ns("2006-01-02T15:04:05Z07:00", rule_data_effective_on(data.rule_data__configuration__)) < time.now_ns()
```

Reply:
There's actually a dedicated Rego function for parsing `rfc3339`: https://www.openpolicyagent.org/docs/latest/policy-reference/#builtin-time-timeparse_rfc3339_ns

I changed my PR to use that.